### DISCUSSION TITLE
Use of Checkbox and Radio

### DISCUSSION THREAD

**Discussion by thejfreitas** [justeattakeaway/pie#1389] [General]
I would like to raise a discussion about the use of checkbox and radio across the organization and maybe make a decision on how we are going to provide these components.

In Canada we provide four different components:
Checkbox: https://react.pie.design/?path=/story/components-checkbox-checkbox--playground
Checkbox Group: https://react.pie.design/?path=/story/components-checkbox-checkboxgroup--playground
RadioButton: https://react.pie.design/?path=/story/components-radio-button-radio-button--playground
RadioButtonGroup: https://react.pie.design/?path=/story/components-radio-button-radio-button-group--playground

I was wondering how we are going to provide these components in web-c

I noticed that the Radio component is not being used at Skip but the RadioButtonGroup because it does not make sense for a radio button containing only one option.

In my opinion we can implement the Checkbox and RadioButton with the capabilities of CheckboxGroup and RadioButtonGroup giving the user to choose how many options they want to render in the Checkbox or RadioButton.

Let me know what you all think about it.

**Comment by ashleynolan** [comment]
I think it's likely worth a chat in our next catchup – it's a really good shout for sure.

In terms of what we have from design currently, we have the [radio button designs](https://www.figma.com/file/pPSC73rPin4csb8DiK1CRr/%E2%9C%A8-%5BCore%5D-Web-Components-%5BPIE-3%5D?type=design&node-id=198-33663&mode=design&t=JI26aVZRGCNUkdOv-0) and some guidance on alignment in terms of groupings [in the component docs](https://www.figma.com/file/j1YKygEyhqZ6zKVxcHapn5/%5BCore%5D-Component-Documentation-%E2%9A%AA%EF%B8%8F-%5BPIE-2.0%5D?type=design&node-id=32007-383175&mode=design&t=RFpwc9XlFTfX43gR-4). It may be better as you say to actually make the grouped usage more explicit by having a `radioButtonGroup` component, as I agree it's unlikely anyone would use a single radio button.

@dandel10n – would be worth getting your thoughts on this, as you're championing the checkbox/radio button components 😄
